src/scatter_buildblock/ScatterEstimationByBin.cxx

```diff
@@ -11,447 +11,1274 @@
+
+            shared_ptr<ProjData> temp_projdata ( new ProjDataInMemory (this->scaled_est_projdata_sptr->get_exam_info_sptr(),
+                                                                       this->scaled_est_projdata_sptr->get_proj_data_info_sptr()));
+            temp_projdata->fill(*this->scaled_est_projdata_sptr);
+
+            pow_times_add min_threshold (0.0f, 1.0f, 1.0f, 1e-9f, NumericInfo<float>().max_value());
+
+            for (int segment_num = temp_projdata->get_min_segment_num();
+                 segment_num <= temp_projdata->get_max_segment_num();
+                 ++segment_num)
+            {
+                SegmentByView<float> segment_by_view =
+                        temp_projdata->get_segment_by_view(segment_num);
+
+                in_place_apply_function(segment_by_view,
+                                        min_threshold);
+
+                if (!(temp_projdata->set_segment(segment_by_view) == Succeeded::yes))
+                    warning("Error set_segment %d\n", segment_num);
+            }
+
+            pow_times_add add_scalar(-1e-9f, 1.0f, 1.0f, NumericInfo<float>().min_value(),
+                                     NumericInfo<float>().max_value());
+
+            for (int segment_num = temp_projdata->get_min_segment_num();
+                 segment_num <= temp_projdata->get_max_segment_num();
+                 ++segment_num)
+            {
+                SegmentByView<float> segment_by_view =
+                        temp_projdata->get_segment_by_view(segment_num);
+
+                in_place_apply_function(segment_by_view,
+                                        add_scalar);
+
+                if (!(temp_projdata->set_segment(segment_by_view) == Succeeded::yes))
+                    warning("Error set_segment %d\n", segment_num);
+            }
+
+            // ok, we can multiply with the norm
+            for (int segment_num = temp_projdata->get_min_segment_num();
+                 segment_num <= temp_projdata->get_max_segment_num();
+                 ++segment_num)
+            {
+                SegmentByView<float> segment_by_view_data =
+                        temp_projdata->get_segment_by_view(segment_num);
+
+                SegmentByView<float> segment_by_view_norm =
+                        this->norm_projdata_2d_sptr->get_segment_by_view(segment_num);
+
+                segment_by_view_data *= segment_by_view_norm;
+
+
+                if (!(temp_projdata->set_segment(segment_by_view_data) == Succeeded::yes))
+                    warning("Error set_segment %d\n", segment_num);
+            }
+
+            std::stringstream convert;   // stream used for the conversion
+            convert << this->o_scatter_estimate_prefix << "_" <<
+                       i_scat_iter+1;
+            std::string output_filename = convert.str();
+
+            BinNormalisationFromProjData normalisation_coeffs(this->norm_projdata_sptr);
```

Comment:
overlap with other norm object?

possibly not easy to fix due to the gaps stuff above. sigh.